### PR TITLE
Allows store a snapshot draft

### PR DESCRIPTION
**Pull request by mirtagonz**
- Create model for the management of partial snapshots.
- Allows to save a partial snapshot.
- Shows survey buttons according to the role.

### CODE REVIEW

**Comment by mirtagonz** [comment]
Change "snapshot_tmp" to "snapshot_draft" to match the names with the backend.

File: src/app/components/form.js [hunk 1]
```diff
@@ -146,36 +146,82 @@ class Form extends Component {
...
+    } else {
+      this.render();
+    }
+  }
+
+  checkShowSaveDraft(state){
+    let show = true;
+
+    if(this.props.uiSchema['ui:group:personal'].length === 0){
+      show = false;
```
Comment (rodrigojv) [inline src/app/components/form.js]: Can we return here a false value? Or we still need to go to the for loop?

File: src/app/components/form.js [hunk 2]
```diff
@@ -146,36 +146,82 @@ class Form extends Component {
...
+    }
+
+    for(let i=0; i<this.props.uiSchema['ui:group:personal'].length; i++){
+      if(!state.formData){
```
Comment (rodrigojv) [inline src/app/components/form.js]: Can we put this if outside the for loop and return a false value if the state does not have a formData?

File: src/app/components/form.js [hunk 3]
```diff
@@ -146,36 +146,82 @@ class Form extends Component {
 
   }
 
+  onSaveDraft(){
+    if(this.checkShowSaveDraft(this.state)){
+      this.props.handleSaveDraft(this.state);
```
Comment (rodrigojv) [inline src/app/components/form.js]: Can we add a comment here explaining why we call `handleSaveDraft`?

File: src/app/components/form.js [hunk 4]
```diff
@@ -146,36 +146,82 @@ class Form extends Component {
...
-              onClick={() => this.onCancel(this.state.formData)}
-              className="btn btn-circle survey-previous"
-            >
-              <i className="fa fa-chevron-left" />
-            </button>
-            <button type="submit" className="btn btn-circle survey-next">
-              <i className="fa fa-chevron-right" />
-            </button>
+      <div>
+        <button className={classButton} id="save-draft" onClick={() => this.onSaveDraft()}> Save Draft </button>
```
Comment (rodrigojv) [inline src/app/components/form.js]: Maybe we can add the logic to show the button here inline with JS:
```javascript
{this.checkShowButton(this.state) && 
<button className="btn btn-primary" ....
}
```

**Comment by mirtagonz** [comment]
@rodrigojv  The requested improvements were made,  can you check please?

**Comment by rodrigojv** [comment]
@mirtagonz could you solve the conflicts? Thank you!

**Comment by mirtagonz** [comment]
@rodrigojv  the conflicts were resolved!

**Pull request merged**